plugins/org.eclipse.sirius.table/src/org/eclipse/sirius/table/business/internal/refresh/DTableElementSynchronizer.java

```diff
@@ -17,22 +17,25 @@
 import java.util.Collections;
```

Comment:
There are a lot of modifications in this class. It's difficult to review without missing anything. I think that a code coverage must be done before and after changes with EclEmma using suite:
org.eclipse.sirius.tests.suite.table.AllTablePluginTests
org.eclipse.sirius.tests.suite.table.AllTableStandaloneTests
and org.eclipse.sirius.tests.swtbot.suite.TableSwtbotTestSuite
If the coverage is not of 100%, I think that additional tests must be added in the context of these changes.

Reply:
Code coverage gives the same level of coverage before and after the refactoring.
Following segments of case were/are not covered:

In CrossTableSynchronizer
- if no interactions, remove any existing cells.
- refreshColumnCandidate when position changes in description, it is reflect in the table.
- remove columns without target.
- Evaluate semantic candidates of IntersectionMapping without domain.
- updateCells: handle mapping change and/or target change.

In AbstractTableSynchronizer
- detectLineToRefresh : moving order of lines.
- remove line without targets.

In EditionTableSynchronizer
- remove cell without lines.
- removeOldCellsInTable : this is probably dead code. there is a comment about it.
- refresh : reach cellStillExists == false.
- refreshColumnMapping : moving columns : adding/removing column.

Reply:
The list of tests to add could be done later. But it's an intersting list to increase the code coverage.